java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -104,8 +176,9 @@ public void initializeChildrenFromFields(List<Field> children) {
 
   @Override
   public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers) {
+    // TODO: Check if need to truncate bits vector
```

Comment:
This should be similar to the code in struct?

Reply:
Struct doesn't do any buffer truncating:
https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/complex/NullableMapVector.java#L94

However, I am not sure about the purpose of buffer truncating here originally (and in other vectors too), maybe @siddharthteotia can help explain?